llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -159,13 +160,23 @@ def gen_rotary_embedding(
         llama_rope_config['rope_type'] = llama_rope_config.pop('type')
         if llama_rope_config['rope_type'] == 'no_scaling':
             llama_rope_config['rope_type'] = 'default'
-        partial_llama_config = PartialLlamaConfig(
-            rope_scaling=llama_rope_config,
-            rope_theta=rope_theta,
-            max_position_embeddings=max_seq_len,
-            hidden_size=d_model,
-            num_attention_heads=n_heads,
-        )
+        if head_dim is not None:
```

Comment:
I don't think you need an if/else here, passing `head_dim` when its `None` is the same as not passing it i believe.